javaslang/src/main/java/javaslang/collection/HashArrayMappedTrie.java

```diff
@@ -101,6 +182,20 @@ static int fromBitmap(int bitmap, int bit) {
 
         abstract AbstractNode<K, V> modify(int shift, int keyHashCode, K key, V value, Action action);
 
+        Iterator<LeafNode<K, V>> nodes() {
+            return new It<>(this);
+        }
+
+        @Override
+        public Iterator<Tuple2<K, V>> iterator() {
+            return nodes().map(node -> Tuple.of(node.key(), node.value()));
```

Comment:
Why is a node not backed by a Tuple?, i.e. s.th. like `nodes().map(AbstractNode::entry)`